core/consensus/validation/babe_block_validator.cpp

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "consensus/validation/babe_block_validator.hpp"
+
+#include <algorithm>
+
+#include <sr25519/sr25519.h>
+#include <boost/assert.hpp>
+#include "crypto/util.hpp"
+#include "scale/scale.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::consensus,
+                            BabeBlockValidator::ValidationError,
+                            e) {
+  using E = kagome::consensus::BabeBlockValidator::ValidationError;
+  switch (e) {
+    case E::NO_AUTHORITIES:
+      return "no authorities are provided for the validation";
+    case E::INVALID_DIGESTS:
+      return "the block does not contain valid BABE digests";
+    case E::INVALID_SIGNATURE:
+      return "SR25519 signature, which is in BABE header, is invalid";
+    case E::INVALID_VRF:
+      return "VRF value and output are invalid";
+    case E::TWO_BLOCKS_IN_SLOT:
+      return "peer tried to distribute several blocks in one slot";
+    case E::INVALID_TXS:
```

Comment:
I think it won't be used too many times and in this case it's better to avoid a short name. To me tx is not so widely used and may be obscure.
```suggestion
    case E::INVALID_TRANSACTIONS:
```